core/pom.xml

```diff
@@ -89,7 +84,6 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <scope>test</scope>
```

Comment:
Baby steps until either we get shaded jars working correctly (which I actually haven't seen any problems with in my local testing of maven-shade-plugin), or until we implement `Types.findGenericParameter` by hand.